core/src/language/go.rs

```diff
@@ -26,38 +28,43 @@ pub struct Go {
 }
 
 impl Language for Go {
-    fn generate_types(&mut self, w: &mut dyn Write, data: &ParsedData) -> std::io::Result<()> {
+    fn generate_types(
+        &mut self,
+        w: &mut dyn Write,
+        _imports: &CrateTypes,
+        data: ParsedData,
+    ) -> std::io::Result<()> {
         // Generate a list of all types that either are a struct or are aliased to a struct.
         // This is used to determine whether a type should be defined as a pointer or not.
-        let mut types_mapping_to_struct = HashSet::new();
-        for s in &data.structs {
-            types_mapping_to_struct.insert(s.id.original.as_str());
-        }
+        let mut types_mapping_to_struct =
+            HashSet::from_iter(data.structs.iter().map(|s| s.id.original.clone()));
+
         for alias in &data.aliases {
-            if types_mapping_to_struct.contains(&alias.r#type.id()) {
-                types_mapping_to_struct.insert(alias.id.original.as_str());
+            if types_mapping_to_struct.contains(alias.r#type.id()) {
+                types_mapping_to_struct.insert(alias.id.original.clone());
```

Comment:
also here, can we avoid clone / use byref

Reply:
I don't think so, but this is cheaper than the clone that happened before. The parsed data will be moved in each call for `write_enum`, `write_struct` and `write_type_alias` so we cant have references in `types_mapping_to_struct`.  The original code cloned all the types: https://github.com/1Password/typeshare/blob/main/core/src/language/go.rs#L46.

Reply:
Actually this may be possible. Working on it.

Reply:
Updated